chain/dependencies.go

```diff
@@ -116,11 +118,6 @@ type Mempool interface {
 
 // TODO: add fixed rules as a subset of this interface
 type Rules interface {
-	// Should almost always be constant (unless there is a fork of
-	// a live network)
-	NetworkID() uint32
-	ChainID() ids.ID
-
```

Comment:
Slightly arbitrary, but `NetworkID` and `ChainID` fit nicely under the concept of rules imo.

Currently, the chainID is only used for replay protection, but this would likely be exposed directly to the VM in the future as well. For example, in the Warp precompile for Subnet-EVM/Coreth, we expose this directly via `getBlockchainID`: https://github.com/avalanche-foundation/ACPs/tree/main/ACPs/30-avalanche-warp-x-evm#precompile-implementation.

Reply:
I agree that the VM needs to know the `NetworkID` and `ChainID`, but I just don't know if `Rules` is the right place to put it. I understand the `Rules` abstraction as a config that exists per fork/time ([ref](https://github.com/ava-labs/hypersdk/blob/bed09991e604939b5dda29ec3e28a2c00d39e717/vm/dependencies.go#L94), this interface signature suggests to me that the point of rules is that it changes over time as time is a parameter). `NetworkID` and `ChainID` are always going to be constant throughout the lifecycle of a VM, so I don't see a reason to bind this to the `Rules` interface since the `Controller` should just be using the `ChainID` and `NetworkID` in `snow.Context` during `Initialize`. Basically I think that this is something that `avalanchego` defines, not something the `Controller` defines.

Reply:
The comment above explains why. If there's a fork of a live network, you may want to switch the networkID/chainID used from the fork point on, so that transactions sent on one network cannot be replayed on the other.